roofit/histfactory/src/HistFactoryNavigation.cxx

```diff
@@ -871,10 +847,10 @@ namespace RooStats {
    // Loop over the sample nodes in this
    // channel's RooRealSumPdf
    RooArgList nodes = sumPdf->funcList();
-   TIter sampleItr = nodes.createIterator();
-   RooAbsArg* sample;
-   while( (sample=(RooAbsArg*)sampleItr.Next()) ) {
-
+   //TIter sampleItr = nodes.createIterator();
+   //RooAbsArg* sample;
+   //while( (sample=(RooAbsArg*)sampleItr.Next()) ) 
+   for (auto const *sample : static_range_cast<RooAbsArg *>(nodes)) {      
```

Comment:
```suggestion
   for (auto * func : static_range_cast<RooAbsReal*>(sumPdf->funcList())) {
```
Here you can do more things in one line: you can already cast to `RooAbsReal*` here to absorb the line below, and replace `sample` with `func` in the code below. You also don't need the temporary `nodes` and the old commented-out lines can be removed.